pkg/s3/client.go

```diff
@@ -0,0 +1,87 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-logr/logr"
+)
+
+// S3Client wraps the AWS SDK client
+type Client struct {
+	s3         s3.Client
+	bucketName string
+	directory  string
+	timeout    time.Duration
+}
+
+type Config struct {
+	BucketName string
+	Region     string
+	Directory  string
+	Timeout    time.Duration
+}
+
+// New initializes a new S3 client
+func New(c Config, ctx context.Context) (*Client, error) {
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(c.Region))
```

Comment:
I have not added any auth here for now since I was only thinking about the public bucket.
Note to self to check it actually works or we need some credentials anyway

Reply:
The [bucket](https://gs-public-capi-vm-images.s3.eu-central-1.amazonaws.com/) is public so it should be fine to leave it as is 👍

Reply:
Yeah technically it should be fine.  But aws can be weird sometimes depending on how you access what